compiler/rustc_mir_build/src/builder/mod.rs

```diff
@@ -866,6 +867,9 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                     HirPlaceBase::Upvar(upvar_id) => upvar_id.var_path.hir_id,
                     _ => bug!("Expected an upvar"),
                 };
+                let upvar_base = upvar_owner.get_or_insert(var_id.owner);
+                assert_eq!(*upvar_base, var_id.owner);
+                let var_id = var_id.local_id;
```

Comment:
can you inline this into its usage?

Reply:
there are no early returns, so I thought it good to do the check and the conversion as early as possible